packages/parameters/src/AppConfigProvider.ts

```diff
@@ -0,0 +1,115 @@
+import { BaseProvider, DEFAULT_PROVIDERS } from './BaseProvider';
+import {
+  AppConfigDataClient,
+  StartConfigurationSessionCommand,
+  GetLatestConfigurationCommand,
+} from '@aws-sdk/client-appconfigdata';
+import type {
+  StartConfigurationSessionCommandInput,
+  GetLatestConfigurationCommandInput,
+} from '@aws-sdk/client-appconfigdata';
+import type { AppConfigGetOptionsInterface } from './types/AppConfigProvider';
+
+class AppConfigProvider extends BaseProvider {
+  public client: AppConfigDataClient;
+  private application: string;
+  private environment: string;
+  private latestConfiguration: Uint8Array | undefined;
+  private token: string | undefined;
+
+  /**
+   * It initializes the AppConfigProvider class'.
+   * *
+   * @param {AppConfigGetOptionsInterface} config
+   */
+  public constructor(options: AppConfigGetOptionsInterface) {
+    super();
+    this.client = new AppConfigDataClient(options.clientConfig || {});
+    this.application = options?.sdkOptions?.application || 'app_undefined'; // TODO: make it optional when we add retrieving from env var
+    this.environment = options?.sdkOptions?.environment || 'env_undefined';
+  }
+
+  public async get(name: string, options?: AppConfigGetOptionsInterface): Promise<undefined | string | Uint8Array | Record<string, unknown>> {
+    return super.get(name, options);
+  }
+
+  /**
+   * Retrieve a parameter value from AWS App config.
+   *
+   * @param {string} name - Name of the configuration
+   * @param {AppConfigGetOptionsInterface} options - SDK options to propagate to `StartConfigurationSession` API call
+   * @returns {Promise<Uint8Array | undefined>}
+   */
+  protected async _get(
+    name: string,
+    options?: AppConfigGetOptionsInterface
+  ): Promise<Uint8Array | undefined> {
+    /**
+     * The new AppConfig APIs require two API calls to return the configuration
+     * First we start the session and after that we retrieve the configuration
+     * We need to store the token to use in the next execution
+     **/
+    if (!this.token) {
+      const sessionOptions: StartConfigurationSessionCommandInput = {
+        ConfigurationProfileIdentifier: name,
+        EnvironmentIdentifier: this.application,
+        ApplicationIdentifier: this.environment,
+      };
+
+      if (options?.sdkOptions) {
+        Object.assign(sessionOptions, options.sdkOptions);
```

Comment:
Here we should apply a similar logic to what done in the `DynamoDBProvider` and `SecretsProvider` ([example](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/5144b3b974bc1f988c2d3d0599f6f05a76cc315c/packages/parameters/src/DynamoDBProvider.ts#L45)) and remove from `sdkOptions` some of the parameters that we don't want the user to override, namely: `ConfigurationProfileIdentifier`, `EnvironmentIdentifier`, and `ApplicationIdentifier`, which leaves us only with [`RequiredMinimumPollIntervalInSeconds`](https://docs.aws.amazon.com/appconfig/2019-10-09/APIReference/API_appconfigdata_StartConfigurationSession.html#appconfig-appconfigdata_StartConfigurationSession-request-RequiredMinimumPollIntervalInSeconds).

Reply:
This way we nudge users to configure the application and environment via the class constructor and use the `name` parameter for the configuration identifier. By disallowing these overrides we can streamline our logic.

Reply:
And for the sake of completion, the `AppConfigGetOptionsInterface` should look like this:
```ts
interface AppConfigGetOptionsInterface extends GetOptionsInterface {
  sdkOptions?: StartConfigurationSessionCommandInput
}
```

Reply:
> This way we nudge users to configure the application and environment via the class constructor and use the `name` parameter for the configuration identifier. By disallowing these overrides we can streamline our logic.

So if a user wants to get configuration with other `environment` and `application`, they should create a new instance of the `AppConfigProvider`. Correct?

Reply:
Yes, that's right.

Looking at the docs for the AppConfig service the two entities are defined as follows:
**Application:**
> In AWS AppConfig, an application is simply an organizational construct like a folder. This organizational construct has a relationship with some unit of executable code. For example, you could create an application called MyMobileApp to organize and manage configuration data for a mobile application installed by your users.

**Environment:**
> An environment is a logical deployment group of AppConfig targets, such as applications in a Beta or Production environment. You can also define environments for application subcomponents such as the Web, Mobile, and Back-end components for your application.

Based on these definitions I think it's safe to assume (and recommend) that users group all their configurations within the same environment & application pair. This has a number of advantages like isolation of both responsibilities and permissions (i.e. this AWS Lambda resource has access to a specific application & environment).

Reply:
> Based on these definitions I think it's safe to assume (and recommend) that users group all their configurations within the same environment & application pair. This has a number of advantages like isolation of both responsibilities and permissions (i.e. this AWS Lambda resource has access to a specific application & environment).

Make sense and simplifies implementation. Thanks for the clarification.